src/domain/repository/salon.js

```diff
@@ -0,0 +1,71 @@
+const Salon = require('../../domain/models/salon');
+
+module.exports = ({ database }) => {
+  const add = async (salon, t) => {
+    const data = toDatabase(salon);
+
+    const new_salon = await database.models.salon.create(data, {
+      transaction: t,
+    });
+    return toDomain(new_salon);
+  };
+
+  const getAll = async (salon, t) => {
+    const new_salon = await database.models.salon.findAll({
+      transaction: t,
+    });
+
+    let salons = new_salon.map((k) => toDomain(k));
+    return salons;
+  };
+
+  const getFew = async (salon, t) => {
+    const new_salon = await database.models.salon.findAll({
+      where: { City: salon.City },
```

Comment:
searchByKey  : city
searchByValue :  Banglore

salon : data , query , payload

Reply:
where :{
[searchByKey] : searchByValue
 }

Reply:
put the data filter

Reply:
done